nimbus-core/src/main/java/com/antheminc/oss/nimbus/support/DefaultLoggingInterceptor.java

```diff
@@ -74,51 +79,60 @@ public String toString() {
 		}
 	}
 	
-	public static String format(String key, String methodName) {
-		return new StringBuilder().append(key).append(methodName).toString();
+	public static String format(String key, int uniqueId, String methodContext) {
+		return new StringBuilder().append(key).append(K_METHOD_EXEC_ID).append(uniqueId).append(K_SPACE).append(methodContext).toString();
+	}
+	
+	public static String format(String key, int uniqueId, String methodName, SimpleStopWatch sw) {
+		return new StringBuilder().append(key).append(K_METHOD_EXEC_ID).append(uniqueId).append(K_SPACE).append(methodName).append(K_SPACE).append(sw.toString()).append(K_MILLI_SECS).toString();
 	}
 	
-	public static String format(String key, String methodName, SimpleStopWatch sw) {
-		return new StringBuilder().append(key).append(methodName).append(K_SPACE).append(sw.toString()).append(K_MILLI_SECS).toString();
+	public static String format(String key, int uniqueId, SimpleStopWatch sw) {
+		return new StringBuilder().append(key).append(K_METHOD_EXEC_ID).append(uniqueId).append(K_SPACE).append(sw.toString()).append(K_MILLI_SECS).toString();
 	}
 
-	public static String format(String key, String methodName, String args) {
-		return new StringBuilder().append(key).append(methodName).append(K_SPACE).append(args).toString();
+	public static String format(String key, int uniqueId, String methodName, String args) {
+		return new StringBuilder().append(key).append(K_METHOD_EXEC_ID).append(uniqueId).append(K_SPACE).append(methodName).append(K_SPACE).append(args).toString();
 	}
 	
+	
 	@Around("@within(com.antheminc.oss.nimbus.support.EnableAPIMetricCollection)")
 	public Object logMethods(ProceedingJoinPoint proceedingJoinPoint) throws Throwable {
 		String methodName = nullSafeMethodGet(proceedingJoinPoint);
 		SimpleStopWatch sw = new SimpleStopWatch();
 		
 		checkAndInjectProxyReferenceToTarget(proceedingJoinPoint);
 		
+		if(methodExecIdInThread.get() == null) {
+			methodExecIdInThread.set(new AtomicInteger());
```

Comment:
ThreadLocal needs to be reset at the end of the call invocation, otherwise when the same thread goes back into the pool and gets reused later, it will continue from where it had the last count.